lib/crawly/engine.ex

```diff
@@ -87,4 +99,19 @@ defmodule Crawly.Engine do
 
     {:reply, msg, %Crawly.Engine{state | started_spiders: new_started_spiders}}
   end
+
+  defp list_all_spider_status(started_spiders) do
+    Crawly.Utils.list_spiders()
```

Comment:
Please move this call to list_spiders() on line 40. As I see it we don't need to do it within the context of the Engine process.

Reply:
I moved it to a priv function as the stop_all_spiders and (I think) start_all_spiders uses it to check if a spider should be started/stopped

Reply:
I was asking to move `Crawly.Utils.list_spiders()` out from the `handle_call` scope. It's potentially a heavy operation (and I saw erlang nodes killed by requests to list all applications). So engine will not be blocked by the listing process.

Another idea which sits in my mind is having a registry of spiders in the Engine. So when engine starts we can do `Crawly.Utils.list_spiders()` so it will know which spiders it have, and will store this info in the state. With the idea that if you want to update the list you can do for example `Engine.scan_spiders/0` or `Engine.add_spider`